engine/src/flutter/impeller/typographer/text_frame.cc

```diff
@@ -101,12 +88,7 @@ void TextFrame::SetPerFrameData(Scalar scale,
                                 Point offset,
                                 const Matrix& transform,
                                 std::optional<GlyphProperties> properties) {
-  if (!transform_.Equals(transform) || !ScalarNearlyEqual(scale_, scale) ||
-      !ScalarNearlyEqual(offset_.x, offset.x) ||
-      !ScalarNearlyEqual(offset_.y, offset.y) ||
-      !TextPropertiesEquals(properties_, properties)) {
-    bound_values_.clear();
-  }
+  bound_values_.clear();
```

Comment:
How is this supposed to work?  Should bounds_values_ just be added to SetFrameData?

Reply:
Essentially what bound_values_ does is it lets us do a single lookup for the glyph locations of glyphs (instead of multiple). When we check if the glyph atlas is populated with a glyph, we stuff the resulting data into bound_values so that the text frame can re-use it without doing another slow hash lookup.

Reply:
We don't know the bound_values until after we populate the glyph atlas, which happens necessarily after preroll when this method is called.